src/NLog.StructuredLogging.Json.Tests/FlattenedJsonCanSerializeToFile.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Newtonsoft.Json.Linq;
+using NLog.Config;
+using NLog.Targets;
+using NUnit.Framework;
+
+namespace NLog.StructuredLogging.Json.Tests
+{
+    [TestFixture]
+    public class FlattenedJsonCanSerializeToFile
+    {
+        [Test]
+        public void JsonOutputIsCorrect()
+        {
+            var fileName = UniqueLogFileName();
+
+            var fileLogger = GetFlattenedJsonFileLogger(fileName);
+            Assert.That(fileLogger, Is.Not.Null);
+
+            fileLogger.ExtendedInfo("This is a test", 
+                new
+                {
+                    Hello = "Hello",
+                    SomeText = "this is some, text. It has! punctation?",
+                    ANumber = 42,
+                    Sometime = new DateTime(2017, 4, 5, 6, 7, 8, DateTimeKind.Utc)                    
+                });
+
+            LogManager.Flush();
+            AssertIsJsonLogEntry(fileName);
+            ClearLoggingConfig();
+        }
+
+        private void AssertIsJsonLogEntry(string fileName)
+        {
+            var json = ParseFile(fileName);
+
+            // it should be json
+            Assert.That(json, Is.Not.Null);
+            Assert.That(json.HasValues);
+
+            // standard props should be present
+            Assert.That(json.GetValue("Message"), Is.Not.Null);
+            Assert.That(json.GetValue("TimeStamp"), Is.Not.Null);
+            Assert.That(json.GetValue("Level"), Is.Not.Null);
+
+            // custom props should be present
+            Assert.That(json.GetValue("Hello"), Is.Not.Null);
+            Assert.That(json.GetValue("SomeText"), Is.Not.Null);
+            Assert.That(json.GetValue("ANumber"), Is.Not.Null);
+            Assert.That(json.GetValue("Sometime"), Is.Not.Null);
+
+            // but not true for any old string
+            Assert.That(json.GetValue("noSuchProp12345asdfg"), Is.Null);
+        }
+
+        private static JObject ParseFile(string fileName)
+        {
+            if (!File.Exists(fileName))
+            {
+                Assert.Fail($"no output file exists with name {fileName}");
+            }
+
+            var contents = File.ReadAllText(fileName);
+
+            if (string.IsNullOrWhiteSpace(contents))
+            {
+                Assert.Fail($"Output file {fileName} has no  contents");
+            }
+
+            // this will throw if the json is not valid
+            return JObject.Parse(contents);
+        }
+
+        private static string UniqueLogFileName()
+        {
+            var unique = Guid.NewGuid().ToString("N");
```

Comment:
Using the dir that I am already working in (typically `C:\Code\NLog.StructuredLogging.Json\src\NLog.StructuredLogging.Json.Tests\bin\Debug\net462` ) was useful while isolating this issue, but now that it is complete, the "temp directory" seems like a better option.